lib/pure/collections/lists.nim

```diff
@@ -461,22 +461,35 @@ proc prepend*[T](L: var SinglyLinkedList[T], value: T) {.inline.} =
 func copy*[T](L: SinglyLinkedList[T]): SinglyLinkedList[T] {.since: (1, 5).} =
   ## Creates a shallow copy of `L`.
   runnableExamples:
-    let a = [1, 2, 3].toSinglyLinkedList
-    assert $a == $a.copy
+    type Foo = ref object
+      x: int
+    var f = Foo(x: 1)
+    let
+      a = [f].toSinglyLinkedList
+      b = a.copy
+    f.x = 42
+    assert a.head.value.x == 42
+    assert b.head.value.x == 42
```

Comment:
add this or similar to emphasize the copy semantics. Avoids any guesswork when reading docs!
(requiers changing a to var)
```nim
a.add [f].toSinglyLinkedList
doAssert a.toSeq == [f, f]
doAssert b.toSeq == [f]
```

Reply:
If you insist... but (at least for me) the immutable argument in the declaration is enough to be sure that `b` is not modified.

Reply:
unfortunately that's not the case, see https://github.com/nim-lang/Nim/issues/16312#issuecomment-742892857

Reply:
I'm not sure I understand ... from that thread I see that an assignment can make a (bad) shallow copy, and mess up things. But surely a `func` such as this `copy` cannot do that?